Siwx91x Compilation error with PM

### Describe the bug

On Siwx91x SoC, a compilation error happens when CONFIG_PM_DEVICE is enabled without CONFIG_PM_DEVICE_RUNTIME and CONFIG_POWER_DOMAIN.

It is because power domain node is missing and needed in order to use PM_DEVICE.

### Regression

- [ ] This is a regression.

### Steps to reproduce

Build a samples where config PM_DEVICE is enabled without CONFIG_PM_DEVICE_RUNTIME and CONFIG_POWER_DOMAIN.

`west build -p -b siwx917_dk2605a/siwg917m111mgtba tests/lib/cpp/cxx -T cpp.main.newlib_nano`

### Relevant log output

```shell
The compilation will that kind of logs:

/opt/toolchains/zephyr-sdk-0.17.4/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/drivers/serial/libdrivers__serial.a(uart_ns16550.c.obj):(.data.__pm_device_dts_ord_45+0x10): undefined reference to `__device_dts_ord_8'
```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

_No response_

### Additional Context

_No response_

--- a/soc/silabs/silabs_siwx91x/Kconfig.defconfig
+++ b/soc/silabs/silabs_siwx91x/Kconfig.defconfig
@@ -13,6 +13,16 @@ configdefault SYS_CLOCK_TICKS_PER_SEC
 configdefault UART_NS16550_DW8250_DW_APB
 	default y
 
+if PM_DEVICE
+
+configdefault PM_DEVICE_RUNTIME
+	default y
+
+configdefault POWER_DOMAIN
+	default y
+
+endif # PM_DEVICE
+
 if SILABS_SIWX91X_NWP
 
 # WiseConnect create threads with realtime priority. Default (10kHz) clock tick

